pgmpy/factors/CPD.py

```diff
@@ -112,6 +112,27 @@ def __init__(self, variable, variable_card, values,
 
         super().__init__(variables, cardinality, values.flatten('C'))
 
+    def replace_variables(self, var1, var2):
+        new_scope = self.scope()
+        new_scope.reverse()
+        new_cardinality = self.cardinality[::-1].copy()
```

Comment:
@kislayabhi I did some tests and found that ```.copy()``` is much faster as compared to the one that I had suggested.

```python
In [1]: import numpy as np
In [2]: a = np.arange(100000)
In [3]: %timeit new_cardinality = np.array([var_card for var_card in a[::-1]])
100 loops, best of 3: 13 ms per loop
In [4]: %timeit new_cardinality = a.copy()
The slowest run took 6.74 times longer than the fastest. This could mean that an intermediate result is being cached 
10000 loops, best of 3: 50.9 µs per loop
```

So I would suggest you to go with ```.copy()```.